libbeat/publisher/queue/queue.go

```diff
@@ -34,6 +35,23 @@ type ACKListener interface {
 	OnACK(eventCount int) // number of consecutively published events acked by producers
 }
 
+//Metrics is a set of basic-user friendly metrics that report the current state of the queue. These metrics are meant to be relatively generic and high-level, and when reported directly, can be comprehensible to a user.
+type Metrics struct {
+	//EventCount is the total events currently in the queue
+	EventCount opt.Uint
+	//ByteCount is the total byte size of the queue
+	ByteCount opt.Uint
+	//ByteLimit is the user-configured byte limit of the queue
+	ByteLimit opt.Uint
```

Comment:
This probably requires us to implement https://github.com/elastic/beats/issues/30069 to get. Doing this for the memory queue might be tricky.

I like these and think they are a much more intuitive way to think about the queue size, but I'm not convinced adding this is more important than a lot of the other work we are already doing.

Reply:
Yah, hence why I made all these `opt` values; the thinking was that some queues might care more about bytes, and others would be more event-based, the `opt` type would allow this one struct to forward whatever metrics types are available.

Reply:
OK cool, they can just stay unimplemented then until we add support for them.

Reply:
I was thinking of these fields as referring to the queue's main buffer, which for the disk queue would mean disk usage. That data is already available in the current code, there's just no one to report it to. The memory queue could leave those fields zero for now, since it doesn't enforce a byte limit.

(We could consider adding metrics about memory overhead outside the main buffer too, but like you point out nobody tracks that yet.)

Reply:
@faec Do you think this is something we should merge now? Should this or a follow-up PR actually add a `Metrics()` method to the actual interfaces?